frontend/alert/components/GroupSuggestion.tsx

```diff
@@ -0,0 +1,357 @@
+    selected: boolean;
+    isChecked: boolean;
+}
+
+const Suggestion = ({
+    onClick,
+    onChangeTimeline,
+    courseCode,
+    title,
+    instructor,
+    selected,
+    isChecked,
+}: SuggestionProps) => {
+    const ref = useRef<HTMLDivElement>(null);
+
+    const [checked, setChecked] = useState(() => isChecked);
+
+    useEffect(() => {
+        setChecked(isChecked);
+    }, [isChecked]);
+
+    // If the suggestion becomes selected, make sure that it is
+    // not fully or partially scrolled out of view
+    useEffect(() => {
+        if (selected && ref.current && ref.current.parentElement) {
+            const { bottom, top } = ref.current.getBoundingClientRect();
+            const { parentElement } = ref.current;
+            const {
+                bottom: parentBottom,
+                top: parentTop,
+            } = parentElement.getBoundingClientRect();
+            if (bottom > parentBottom) {
+                parentElement.scrollBy({ top: bottom - parentBottom });
+            } else if (top < parentTop) {
+                parentElement.scrollBy({ top: top - parentTop });
+            }
+        }
+    }, [selected, ref]);
+
+    return (
+        <DropdownItemBox selected={selected} ref={ref}>
+            <CheckboxContainer
+                onClick={() => {
+                    onClick();
+                    setChecked(!checked);
+                }}
+            >
+                <Checkbox checked={checked} />
+            </CheckboxContainer>
+            <DropdownItemLeftCol
+                onClick={() => {
+                    onClick();
+                    setChecked(!checked);
+                }}
+            >
+                <SuggestionTitle>{courseCode}</SuggestionTitle>
+                <SuggestionSubtitle>{instructor || "TBA"}</SuggestionSubtitle>
+            </DropdownItemLeftCol>
+            <IconContainer onClick={onChangeTimeline}>
+                <HistoryIcon icon={faHistory} className="historyIcon" />
+            </IconContainer>
+        </DropdownItemBox>
+    );
+};
+
+interface GroupSuggestionProps {
+    sections: any;
+    courseCode: string;
+    value: string;
+    selectedCourses: Set<Section>;
+    inputRef: React.RefObject<HTMLInputElement>;
+    setValue: (v: any) => void;
+    setTimeline: React.Dispatch<React.SetStateAction<string | null>>;
+    setSelectedCourses: any;
+}
+
+const GroupSuggestion = ({
+    sections,
+    courseCode,
+    value,
+    selectedCourses,
+    inputRef,
+    setValue,
+    setTimeline,
+    setSelectedCourses,
+}: GroupSuggestionProps) => {
+    const initializeButtonStates = () => {
+        const states = {};
+        Object.keys(sections).map((key) => {
+            states[key] = false;
+        });
+
+        return states;
+    };
+
+    const [buttonStates, setButtonStates] = useState(initializeButtonStates);
+
+    /**
+     * Takes the activity/activity of a section and return if all sections of {activity} have been selected
```

Comment:
activity/activity?

Reply:
Originally named course activity as course type but then decided it would be confusing so I replaced all. This is probably a mistake from replacing all type to activity.